### ISSUE TITLE
the implement of the guassian mixture model

### ISSUE THREAD

**Issue by shenyuanyuan** [aakhundov/tf-example-models#1]
Hi,
Which article does the code implementation of guassian mixture model refer to? I mean the process of evaluating the maximum likelihood? There are two items of likelihood, i.e., tf.reduce_sum(tf.log(exp_log_shifted_sum)) and tf.reduce_sum(log_shift). From the results I run the code, the first item (i.e., tf.reduce_sum(tf.log(exp_log_shifted_sum))) always give a nearly zero value. And the second item (i.e., tf.reduce_sum(log_shift)) is really useful. I'm not sure that I obtain the right results. I also can not understand the implementation of this part. So would you tell me which article of the method refer to? Or would you explain the process briefly?

**Comment by shenyuanyuan** [comment]
I understand now. Just like the softmax loss. Before exp operation, the maximum item are subtracted to prevent the overflow
